### PR TITLE
Connect to API

### PR DESCRIPTION
**Pull request by tobuya**
- Use `fetch` with `createAsyncThunk` to make API call
- Fetch booklist from Bookstore API on page load
- Refactor `add book` and `remove book` features to persist changes in the server

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,42 +1,59 @@
...
 
-export const addBookAction = (newBook) => ({ type: ADDED_BOOK, newBook });
-export const removeBookAction = (id) => ({ type: REMOVED_BOOK, id });
+const deleteBook = createAsyncThunk(
+  REMOVED_BOOK,
+  async (id, { dispatch }) => {
+    await fetch(`${appResourceUrl}/${id}`, { method: 'DELETE' });
+    dispatch({ type: REMOVED_BOOK, payload: id });
+  },
+);
```
Comment (Qoosim) [inline src/redux/books/books.js:47]: - [ ] Wonderful job so far. However, there's an issue with the app at this moment. The expected behaviour of the application is to change the state of the app once an action is performed, not by performing double actions. What I am saying is this. When delete button is clicked the expected behaviour is for the state to change from the redux store. But the actual action happens when the page  is refreshed. Kindly look into it and fix.

[page-behaviour.webm](https://user-images.githubusercontent.com/26597432/216650146-b0cd9498-afa4-4d4b-b4c0-6c8537edec8c.webm)

File: src/redux/books/books.js [hunk 2]
```diff
@@ -1,42 +1,59 @@
...
-      return {
-        ...state,
-        books: [...state.books.filter((book) => book.id !== action.id)],
-      };
-    default:
-      return state;
+  if (action.type === FETCHED_BOOK) {
+    return action.payload;
+  } if (action.type === ADDED_BOOK) {
+    return [...state, action.payload];
```
Comment (Qoosim) [inline src/redux/books/books.js:53]: - [ ] I noticed `REMOVE_BOOK` is not added to the bookReducer. You might need to check here for the issue.

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @tobuya, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Git Flow. :heavy_check_mark: 




## Required Changes ♻️

- [ ] The MIT license of your repository is not properly set up. Kindly follow this [link](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository) to properly set it up for your project.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 2 above)
Comment (tobuya) [reply src/redux/books/books.js:53]: @Qoosim Thank you for that. I never realized it :smile:

File: src/redux/books/books.js (hunk 1 above)
Comment (tobuya) [reply src/redux/books/books.js:47]: @Qoosim Thank you so much.

**Review by topeogunleye — APPROVED** [review]
### Hi @tobuya, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Followed Github flow :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good description :heavy_check_mark:
 
## [Optional] suggestions:
* Nothing to mention.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**